packages/web-wallet/src/components/Wallet/AddFunds/AddFunds.js

```diff
@@ -31,22 +32,26 @@ class AddFunds extends Component {
     }
 
     return {
+      ...state,
       balance: props.balance,
       delta: props.balance - state.balance,
     };
   }
 
   componentDidMount() {
-    this.resetPathPayment();
+    this.props.resetRequest('pathPayment');
   }
 
-  resetPathPayment = () => {
+  resetForm = () => {
     this.props.resetRequest('pathPayment');
+    this.setState({ delta: 0 });
   };
 
   getAssetName = () =>
     this.props.match.params.asset === 'native' ? 'XLM' : 'MOBI';
 
+  onTabChange = currentTabIndex => this.setState({ currentTabIndex });
```

Comment:
Why do we want to change the behavior of tabs switch logic? It was designed to encapsulate tabs switch inside `Tabs` component itself

Reply:
It's still encapsulated. `onTabChange` is an optional callback to notify the parent component about the change. Here we use it to preserve selected tab after rendering confirmation message.